ginivision/src/androidTest/java/net/gini/android/vision/review/ReviewScreenTest.java

```diff
@@ -416,4 +415,113 @@ public void onProceedToAnalysisScreen(@NonNull final Document document) {
         assertAbout(document()).that(documentToAnalyze.get()).hasSameUUIDinUserCommentAs(
                 documentToProceedWith.get());
     }
+
+    @Test
+    public void should_returnDocument_withZeroRotationDelta_inAnalyzeDocument()
+            throws InterruptedException {
+        final ReviewActivityTestStub activity = startReviewActivity(TEST_JPEG, 180);
+
+        final AtomicReference<Document> documentToAnalyze = new AtomicReference<>();
+
+        activity.setListenerHook(new ReviewActivityTestStub.ListenerHook() {
+            @Override
+            public void onShouldAnalyzeDocument(@NonNull Document document) {
+                documentToAnalyze.set(document);
+            }
+        });
+
+        // Allow the activity to run a little for listeners to be invoked
+        Thread.sleep(PAUSE_DURATION);
+
+        assertAbout(document()).that(documentToAnalyze.get()).hasRotationDeltaInUserComment(0);
+    }
+
+    @Test
+    public void should_returnDocument_withNonZeroRotationDelta_inProceedToAnalysis_ifDocumentWasRotated()
```

Comment:
Yeah, debugging might be an issue. These are white box Activity integration tests (or white box Activity UI tests) to verify that the library's Activities fulfil our public API contract.

Back when we created the library we implemented these tests to verify that the intended behaviour is fulfilled.

You are right, ideally we would need to cover the behaviour of the library's public API in unit tests and these high-level tests should serve the role of API contract acceptance testing.